backend/tests/services/knowledge/test_knowledge_service.py

```
@@ -332,3 +339,44 @@ def test_delete_document_treats_deleted_status_as_success(self) -> None:
+    def test_cancels_pending_tasks_before_closing_loop(self) -> None:
+        cancelled = threading.Event()
+
+        async def background_task() -> None:
+            try:
+                await asyncio.sleep(3600)
+            except asyncio.CancelledError:
+                cancelled.set()
+                raise
+
+        async def main_coro() -> str:
+            asyncio.create_task(background_task())
```

Ruff (0.15.10)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)